extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/health/KafkaStreamsHealthCheck.java

```diff
@@ -0,0 +1,58 @@
+package io.quarkus.kafka.streams.runtime.health;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Readiness;
+import org.jboss.logging.Logger;
+
+import io.quarkus.kafka.streams.runtime.KafkaStreamsTopologyManager;
+
+@Readiness
+@ApplicationScoped
+public class KafkaStreamsHealthCheck implements HealthCheck {
+
+    private static final Logger LOGGER = Logger.getLogger(KafkaStreamsHealthCheck.class.getName());
+
+    @ConfigProperty(name = "quarkus.kafka-streams.topics")
+    public List<String> topics;
+
+    @ConfigProperty(name = "quarkus.kafka-streams.bootstrap-servers")
+    public List<String> bootstrapServers;
+
+    @Inject
+    private KafkaStreamsTopologyManager manager;
+
+    private String commaSeparatedBootstrapServersConfig;
+    private List<String> trimmedTopics;
+
+    @PostConstruct
+    public void init() {
+        commaSeparatedBootstrapServersConfig = String.join(",", (bootstrapServers));
+        trimmedTopics = topics.stream().map(String::trim).collect(Collectors.toList());
+    }
+
+    @Override
+    public HealthCheckResponse call() {
+        HealthCheckResponseBuilder builder = HealthCheckResponse.named("Kafka topic(s) health check").up();
+
+        try {
+            trimmedTopics.forEach(topic -> builder.withData(topic, true));
+            manager.getMissingTopics(trimmedTopics, commaSeparatedBootstrapServersConfig).forEach(
+                    missingTopic -> builder.down().withData(missingTopic, false));
```

Comment:
And one line with all the missing topics ...

Reply:
I agree with you that one line per topic can be too much.

But IMHO, I think it would be better if we respond with 2 lists:
```
{
    "Available topic(s)" : "topic_1,topic_2,...",
    "Missing topic(s)": "topic_a,topic_b,topic_c,...",
}
```

Because this following example seems wird to me and not very relevant:
```
{
    "Available topic(s)": "topic_1,topic_2,...",
    "topic_a": false,
    "topic_b": false,
    "topic_c": false,
    ...
}
```